x/rollapp/types/liveness.go

```diff
@@ -0,0 +1,67 @@
+package types
+
+import (
+	"bytes"
+	"encoding/binary"
+)
+
+var (
+	LivenessEventQueueKeyPrefix = []byte("LivenessEventQueue")
+	LivenessEventQueueSlash     = []byte("s")
+	LivenessEventQueueJail      = []byte("j")
+)
+
+func LivenessEventQueueKey(e LivenessEvent) []byte {
+	v := LivenessEventQueueSlash
+	if e.IsJail {
+		v = LivenessEventQueueJail
+	}
+	return createLivenessEventQueueKey(&e.HubHeight, v, &e.RollappId)
+}
+
+func LivenessEventQueueIterKey(height *int64) []byte {
+	return createLivenessEventQueueKey(height, nil, nil)
+}
+
+// can be called with no arguments to retrieve all items
+// can be called with only a height, to iterate all events for a height
+// otherwise must have all three arguments, for put/del ops
+func createLivenessEventQueueKey(height *int64, kind []byte, rollappID *string) []byte {
+	if height == nil && (0 < len(kind) || rollappID != nil) {
+		panic("must provide a height")
+	}
+	var key []byte
+	key = append(key, LivenessEventQueueKeyPrefix...)
+	if height != nil {
```

Comment:
u'll have a key without height? breaks the key expected structure

Reply:
no it doesn't
because the key will be used for iteration

Reply:
in what scenario u iterate over all heights?

Reply:
export genesis

Reply:
I would avoid all these conditionals and just split it into functions, each for their own form of the key. It will be easier to read and less error prone

Reply:
the two callsites are right above https://github.com/dymensionxyz/dymension/blob/fb107d860109c07da6a31da5b13004e1b10c067f/x/rollapp/types/liveness.go#L14-L24